diff -Nur maildrop-2.9.3/libs/maildrop/lexer.h maildrop-2.9.3-new/libs/maildrop/lexer.h
--- maildrop-2.9.3/libs/maildrop/lexer.h	2013-08-25 14:54:20.000000000 -0400
+++ maildrop-2.9.3-new/libs/maildrop/lexer.h	2018-05-29 16:12:28.823045565 -0400
@@ -37,9 +37,11 @@
 
 	// nextchar() is used to actually read the next character.
 
-	int	nextchar() { int c=file.get();
+       int     nextchar() { int c=file.get(), incr=0;
 
-				if (c == '\n')	++linenum;
+                               if (c == '\r') { incr=1; c = file.get(); }
+                               if (c == '\n') { incr=1; }
+                               if (incr) { ++linenum; } 
 				return (c);
 			}
 	void	error(const char *);
diff -Nur maildrop-2.9.3/libs/maildrop/main.C maildrop-2.9.3-new/libs/maildrop/main.C
--- maildrop-2.9.3/libs/maildrop/main.C	2017-08-19 09:30:56.000000000 -0400
+++ maildrop-2.9.3-new/libs/maildrop/main.C	2018-05-29 16:14:55.263282625 -0400
@@ -303,9 +303,10 @@
 	}
 
 	uid_t u;
-	if (auth->sysusername)
+	if (auth->sysusername || auth->address)
 	{
-		struct	passwd *q=getpwnam(auth->sysusername);
+		struct	passwd *q = auth->sysusername ?
+			getpwnam(auth->sysusername) : getpwnam(auth->address);
 
 		if (q == NULL)
 		{
@@ -318,8 +319,9 @@
 
 		u=q->pw_uid;
 	}
-	else
+	else {
 		u=*auth->sysuserid;
+	}
 
 	if (VerboseLevel() > 1)
 	{
@@ -387,12 +389,12 @@
 			    void *void_arg)
 {
 	Maildrop &maildrop=*(Maildrop *)void_arg;
-
-	if (VerboseLevel() > 1)
+	//auth.sysusername = &auth->address;
+	if ((VerboseLevel() > 1))
 	{
 		Buffer b;
 
-		b.set(auth->sysgroupid);
+		b.set(auth->sysusername);
 		b.push(0);
 
 		merr << "maildrop: dovecotauth: groupid="
@@ -402,24 +404,31 @@
 	setgroupid(auth->sysgroupid);
 
 	uid_t u;
-	if (auth->sysusername)
+	Buffer name,val;
+	name="RECIPIENT";
+	if (auth->address)
 	{
-		struct	passwd *q=getpwnam(auth->sysusername);
+		struct	passwd *q=getpwnam(auth->address);
 
 		if (q == NULL)
 		{
 			merr << "Cannot find system user "
-			     << auth->sysusername
+			     << auth->address
 			     << "\n";
 
 			nochangeuidgid();
 		}
 
 		u=q->pw_uid;
+		val=auth->address;
 	}
-	else
+	else 
+	{
 		u=*auth->sysuserid;
-
+		val="nobody";
+	}
+	SetVar(name, val);
+	
 	if (VerboseLevel() > 1)
 	{
 		Buffer b;
@@ -885,7 +894,7 @@
 			{
 				if ( !S_ISDIR(buf.st_mode))
 					tempfail("$HOME/.mailfilters should be a directory.");
-				if ( buf.st_mode & (S_IRWXO|S_IRWXG))
+				if ( buf.st_mode & (S_IRWXO|S_IWGRP))
 					tempfail("Invalid permissions on $HOME/.mailfilters - remove world and group perms.");
 				if ( buf.st_uid != getuid())
 					tempfail("Invalid user ownership of $HOME/.mailfilters.");
@@ -1080,7 +1089,7 @@
 
 		if (!S_ISREG(stat_buf.st_mode))
 			tempfail("mailfilter file isn't a regular file.");
-		if (stat_buf.st_mode & (S_IRWXO | S_IRWXG))
+		if (stat_buf.st_mode & (S_IWGRP | S_IWOTH))
 			tempfail("Cannot have world/group permissions on the filter file - for your own good.");
 		if (stat_buf.st_uid != getuid())
 			tempfail("mailfilter file is owned by the wrong user.");
