--- maildrop-3.0.8/libs/maildrop/main.C	2023-04-06 17:54:51.000000000 -0500
+++ maildrop-3.0.8/libs/maildrop/main.C	2025-03-27 13:06:56.777704369 -0500
@@ -314,23 +314,28 @@
 	}
 
 	uid_t u;
+	if (auth->address)
-	if (auth->sysusername)
 	{
+		std::string val, name;
+		struct	passwd *q = getpwnam(auth->address);
+		name = "RECIPIENT";
+		val = auth->address;
-		struct	passwd *q=getpwnam(auth->sysusername);
 
 		if (q == NULL)
 		{
 			merr << "Cannot find system user "
+			     << auth->address
-			     << auth->sysusername
 			     << "\n";
 
 			nochangeuidgid();
 		}
 
 		u=q->pw_uid;
+		SetVar(name, val);
 	}
+	else {
-	else
 		u=*auth->sysuserid;
+	}
 
 	if (VerboseLevel() > 1)
 	{
@@ -903,7 +908,7 @@
 			{
 				if ( !S_ISDIR(buf.st_mode))
 					tempfail("$HOME/.mailfilters should be a directory.");
+				if ( buf.st_mode & (S_IRWXO|S_IWGRP))
-				if ( buf.st_mode & (S_IRWXO|S_IRWXG))
 					tempfail("Invalid permissions on $HOME/.mailfilters - remove world and group perms.");
 				if ( buf.st_uid != getuid())
 					tempfail("Invalid user ownership of $HOME/.mailfilters.");
@@ -1102,7 +1107,7 @@
 
 		if (!S_ISREG(stat_buf.st_mode))
 			tempfail("mailfilter file isn't a regular file.");
+		if (stat_buf.st_mode & (S_IWGRP | S_IWOTH))
-		if (stat_buf.st_mode & (S_IRWXO | S_IRWXG))
 			tempfail("Cannot have world/group permissions on the filter file - for your own good.");
 		if (stat_buf.st_uid != getuid())
 			tempfail("mailfilter file is owned by the wrong user.");
